tests/cross/src/test/scala/tests/pc/TypeDefinitionSuite.scala

```diff
@@ -0,0 +1,39 @@
+package tests.pc
+
+object TypeDefinitionSuite extends BaseTypeDefinitionSuite {
+
+  check("val")(
+    """
+      |class Main(i: Int) {
+      |
+      |}
+      |
+      |object Main {
+      |  val ts@@t = new Main(2)
+      |}""".stripMargin,
+    "Main.scala [2:6 -> 2:9]"
+  )
+
+  check("constructor")(
+    """
+      |class Main(i: Int) {}
+      |
+      |object Main {
+      | def tst(m: Main): Unit = {}
+      |
+      |  tst(new M@@ain(2))
+      |}""".stripMargin,
+    "Main.scala [2:6 -> 2:9]"
+  )
+
+  check("method")(
+    """
+      |object Main {
+      |  def tst(): Unit = {}
+      |
+      |  ts@@t()
+      |}""".stripMargin,
+    "Main.scala [3:6 -> 3:8]"
+  )
```

Comment:
Some test cases to consider
```scala
List(1).hea@@d // should go to `Int.scala` instead of `def head`

class F@@oo(val x: Int) // should go to `Foo`

va@@l x = 42 // should not go anywhere

val x = 4@@2
// can go to Int.scala, but it's OK to not go anywhere either IMO.
// Hover doesn't show types for literals because it gets noisy.

for {
  x <- List(1)
  i@@f x > 1 // I'm not sure what's the expected behavior
} println(x)

"".stripS@@uffix("foo") // should go to Predef.scala

def foo[T](param: T): T = para@@m // should go to `T` type parameter

```

Reply:
Shouldn't ```"".stripS@@uffix("foo")``` go to String?

Reply:
I think it should since, that method is not define din Predef.